### PR TITLE
feat!: add `sections` parameter to template insights

### PR DESCRIPTION
**Pull request by mtojek**
Related #9495 

This PR adds `sections` parameter to template insights. The idea is to shrink the endpoint response, and reduce the number of DB calls.

Changes:
* API breaking change: `report` and `interval_reports` can be omitted in `/insights/templates`
* expose `sections` HTTP parameter for `/insights/templates` (values: `report`, `interval_reports`)
* golden files for testing purposes
* FE adjustments to allow for undefined `report`

### CODE REVIEW

File: coderd/insights.go [hunk 1]
```diff
@@ -289,7 +293,7 @@ func (api *API) insightsTemplates(rw http.ResponseWriter, r *http.Request) {
 	// overhead from a transaction is not worth it.
 	eg.Go(func() error {
 		var err error
-		if interval != "" {
+		if interval != "" && slices.Contains(sections, codersdk.TemplateInsightsSectionIntervalReports) {
```
Comment (mafredri) [inline coderd/insights.go:296]: Should we error if given interval without correct section?

**Review by BrunoQuaresma — APPROVED** [review]

**Review by mafredri — APPROVED** [review]
LGTM! Just a few small comments.

File: site/src/pages/TemplatePage/TemplateInsightsPage/TemplateInsightsPage.tsx [hunk 2]
```diff
@@ -105,11 +106,11 @@ export const TemplateInsightsPageView = ({
         <UserLatencyPanel data={userLatency} />
         <TemplateUsagePanel
           sx={{ gridColumn: "span 3" }}
-          data={templateInsights?.report.apps_usage}
+          data={templateInsights?.report?.apps_usage}
```
[the remark this answers is not in the document]
Comment (mtojek) [reply site/src/pages/TemplatePage/TemplateInsightsPage/TemplateInsightsPage.tsx:109]: I will leave it for @BrunoQuaresma to polish the FE side.

File: codersdk/insights.go [hunk 3]
```diff
@@ -38,6 +38,33 @@ const (
...
+type TemplateInsightsSection string
+
+// TemplateInsightsSection enums.
+const (
+	TemplateInsightsSectionIntervalReports TemplateInsightsSection = "interval_reports"
+	TemplateInsightsSectionReport          TemplateInsightsSection = "report"
+)
+
+// TemplateInsightsSectionAsStrings converts section enums into a slice of strings.
+func TemplateInsightsSectionAsStrings(sections ...TemplateInsightsSection) []string {
```
[the remark this answers is not in the document]
Comment (mtojek) [reply codersdk/insights.go]: Done

File: coderd/insights.go (hunk 1 above)
Comment (mtojek) [reply coderd/insights.go:297]: Added in `parseTemplateInsightsSections`

**Pull request merged**